mmv1/third_party/cai2hcl/common/utils.go

```diff
@@ -63,3 +75,54 @@ func hashicorpCtyTypeToZclconfCtyType(t hashicorpcty.Type) (cty.Type, error) {
 	}
 	return ret, nil
 }
+
+func NewConfig() *transport_tpg.Config {
+	return &transport_tpg.Config{}
+}
+
+// normalizeNodeRec traverses the output map recursively, removes fields which are
+// not a part of TF schema and converts unmarshallable "schema.Set" objects to arrays.
+func normalizeNodeRec(node interface{}, mapSchema map[string]*schema.Schema) interface{} {
```

Comment:
What would you think about recursing primarily based on the schema rather than the asset data? That might be more straightforward, especially in terms of identifying and handling nested objects.

Reply:
Thats a great idea, thanks, worked like a charm and I don't need to manually skip values without schema, love it! Thanks! :)